### PR TITLE
Refactoring to use single array for "CAMERA" "GALLERY" and "WEB" images

### PR DESCRIPTION
**Pull request by sudoDeznit**
- This PR refactors to use single array for "CAMERA" "GALLERY" and "WEB" images instead of 3 different .
- Added new reducer and action images_reducer / images_action that handles the single array of images.
- All the functionality is same -- just removed duplicate code used for 3 time for different types to 1.

**NOTE**
- As this will be a big change we will merge it to new branch `dev` and after proper testing will merge to `staging`.
- Previous Pending PR's should be merged to `staging` before merging `dev` to staging

### CODE REVIEW

File: actions/web_actions.js [hunk 1]
```diff
@@ -1,30 +1,6 @@
-import {
-    ADD_TAGS_TO_WEB_IMAGE,
-    LOAD_MORE_WEB_IMAGES,
-    INCREMENT_WEB_IMAGES_UPLOADED,
-    REMOVE_TAG_FROM_WEB_IMAGE,
-    REMOVE_WEB_IMAGE,
-    WEB_IMAGES,
-    URL
-} from './types';
+import { ADD_IMAGE, DELETE_IMAGE, LOAD_MORE_WEB_IMAGES, URL } from './types';
```
Comment (xlcrr) [inline actions/web_actions.js:1]: Please use multiline

File: actions/web_actions.js [hunk 2]
```diff
@@ -48,20 +24,14 @@ export const checkForImagesOnWeb = token => {
...
 
-                    // if photos is null, pass empty array
                     dispatch({
-                        type: WEB_IMAGES,
+                        type: ADD_IMAGE,
                         payload: {
-                            count: resp.data.count,
-                            photos
+                            images: photos,
+                            type: 'WEB'
```
Comment (xlcrr) [inline actions/web_actions.js:34]: lowercase is fine

File: actions/web_actions.js [hunk 3]
```diff
@@ -112,31 +82,12 @@ export const loadMoreWebImages = (token, photo_id) => {
...
-    };
-};
-
 /**
  * Delete selected web image
  * web image - image that are uploaded from web but not tagged
  */
-export const deleteSelectedWebImages = (token, photoId) => {
+export const deleteSelectedWebImages = (token, photoId, id) => {
+    console.log(token, photoId, id);
```
Comment (xlcrr) [inline actions/web_actions.js:90]: remove log

File: actions/web_actions.js [hunk 4]
```diff
@@ -149,8 +100,8 @@ export const deleteSelectedWebImages = (token, photoId) => {
         })
             .then(resp => {
                 dispatch({
-                    type: REMOVE_WEB_IMAGE,
-                    payload: photoId
+                    type: DELETE_IMAGE,
```
Comment (xlcrr) [inline actions/web_actions.js:103]: remove_web_image and delete_image are different ?

File: screens/gallery/GalleryScreen.js [hunk 5]
```diff
@@ -209,7 +200,7 @@ class GalleryScreen extends Component {
                                         }}>
                                         <Icon
                                             name="ios-checkmark-outline"
-                                            size={24}
+                                            size={20}
```
Comment (xlcrr) [inline screens/gallery/GalleryScreen.js:203]: Reminder to myself: check this

File: screens/gallery/galleryComponents/AlbumList.js [hunk 6]
```diff
@@ -43,8 +43,8 @@ class AlbumList extends Component {
 const mapStateToProps = state => {
     return {
         imagesLoading: state.gallery.imagesLoading,
-        gallery: state.gallery.gallery,
-        geotaggedImages: state.gallery.geotaggedImages
+        geotaggedImages: state.gallery.geotaggedImages,
+        images: state.images.images
```
Comment (xlcrr) [inline screens/gallery/galleryComponents/AlbumList.js:47]: what about `state.images.imagesArray`

**Review by xlcrr — CHANGES_REQUESTED** [review]
Partial review done

File: actions/images_action.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    TOGGLE_SELECTED_IMAGES
+} from './types';
+
+/**
+ * action to add images to state
+ * @param {Array} images
+ * @param {('CAMERA' | 'GALLERY' | 'WEB')} type
+ */
+
+export const addImage = (images, type) => {
```
[the remark this answers is not in the document]
Comment (sudoDeznit) [reply actions/images_action.js:20]: It adds images to the Images array (single array) that we are using now

File: actions/images_action.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    };
+};
+
+/**
+ * Add tags to images
+ */
+
+export const addTagsToImages = payload => {
+    return {
+        type: ADD_TAGS_TO_IMAGE,
```
[the remark this answers is not in the document]
Comment (sudoDeznit) [reply actions/images_action.js:34]: I will change it

File: actions/images_action.js [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+        type: DECREMENT_SELECTED
+    };
+};
+
+/**
+ * delete image by id
+ * @param {number} id
+ */
+
+export const deleteImage = id => {
```
[the remark this answers is not in the document]
Comment (sudoDeznit) [reply actions/images_action.js:53]: I can probably configure .prettierrc.js to do that... will try and create a new PR for that

File: actions/web_actions.js (hunk 1 above)
Comment (sudoDeznit) [reply actions/web_actions.js:1]: This done automatically based on .prettierrc.js config... 
I will configure it to make this multiline and create a new PR for that...

File: actions/web_actions.js (hunk 4 above)
Comment (sudoDeznit) [reply actions/web_actions.js:103]: DELETE_IMAGE is the new type used in image_reducer.
it deletes the image data from state

File: screens/gallery/galleryComponents/AlbumList.js (hunk 6 above)
Comment (sudoDeznit) [reply screens/gallery/galleryComponents/AlbumList.js:47]: the name of images array in the state is `images`

File: actions/web_actions.js (hunk 2 above)
Comment (sudoDeznit) [reply actions/web_actions.js:34]: I was using same convention as we do for redux types..
Easier to identify that it's a type.

can change it to lowercase if u want.

File: screens/gallery/galleryComponents/AlbumList.js (hunk 6 above)
Comment (xlcrr) [reply screens/gallery/galleryComponents/AlbumList.js:47]: I know that, but I always find `images.images` confusing. What do you think about `images.imagesArray`?

File: reducers/images_reducer.js [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+            /**
+             * delete image by id
+             */
+
+            case DELETE_IMAGE:
+                const index = draft.images.findIndex(
+                    image => image.id === action.payload
+                );
+                if (index !== -1) draft.images.splice(index, 1);
+                break;
```
Comment (xlcrr) [inline reducers/images_reducer.js:139]: A gap before every break and return please

File: screens/gallery/GalleryScreen.js [hunk 11]
```diff
@@ -59,8 +59,8 @@ class GalleryScreen extends Component {
     async splitIntoRows(images) {
         let temp = {};
         images.map(image => {
-            const timestampOfImage = image.timestamp * 1000;
-            const placeInTimeOfImage = placeInTime(timestampOfImage);
+            const dateOfImage = image.date * 1000;
```
Comment (xlcrr) [inline screens/gallery/GalleryScreen.js:62]: Do we need to multiply this by 1000, or was that a timestamp only thing?

File: screens/home/HomeScreen.js [hunk 12]
```diff
@@ -476,135 +373,52 @@ class HomeScreen extends PureComponent {
...
-        this.props.webPhotos.map(item => {
-            if (item.tags && Object.keys(item.tags)?.length > 0) {
-                webCount++;
-            }
-        });
-
-        const total = galleryCount + photosCount + webCount;
-
-        console.log({ total });
+        console.log({ imagesCount });
```
Comment (xlcrr) [inline screens/home/HomeScreen.js:384]: remove log?

**Review by xlcrr — APPROVED** [review]
Incredible refactor, well done

File: actions/images_action.js [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+
+/**
+ * Increment the amount of photos selected for deletion
+ */
+
+export const incrementSelected = () => {
+    return {
+        type: INCREMENT_SELECTED
+    };
+};
```
[the remark this answers is not in the document]
Comment (sudoDeznit) [reply actions/images_action.js:87]: We probably can will have a look.

**Pull request merged**